IO.Eventuate.Tram.Tests/Messages/MessageTest.cs

```diff
@@ -0,0 +1,176 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using NSubstitute;
+using Microsoft.EntityFrameworkCore;
+
+namespace IO.Eventuate.Tram.Tests
+{
+
+    [TestClass]
+    public class MessageTest
+    {
+        public string uniqueId = System.DateTime.Now.Ticks.ToString();
+        public string subscriberId = "";
+        public string destination = "";
+        public string payload = "";
+        public DatabaseMessageProducer messageProducer;
+        private List<IO.Eventuate.Tram.Messaging.Common.Message> queue = new List<IO.Eventuate.Tram.Messaging.Common.Message>();
+        public ISet<string> channels = new HashSet<string>();
+        public MessageTest()
+        {
+            uniqueId = System.DateTime.Now.Ticks.ToString();
+            subscriberId = "subscriberId" + uniqueId;
+            destination = "destination" + uniqueId;
+            payload = "Hello" + uniqueId;
+        }
+
+        public class MessageProducer : DatabaseMessageProducer
+        {
+            public MessageProducer(IEnumerable<IMessageInterceptor> messageInterceptors, IIdGenerator idGenerator, IEventuateTramDbContextProvider dbContextProvider, ILogger<DatabaseMessageProducer> logger)
+                : base(messageInterceptors, idGenerator, dbContextProvider,
+                logger)
+            {
+
+            }
+        }
+        public class MessageConsumer : KafkaMessageConsumer
+        {
+            public MessageConsumer(string bootstrapServers,
+            EventuateKafkaConsumerConfigurationProperties eventuateKafkaConsumerConfigurationProperties,
+            DecoratedMessageHandlerFactory decoratedMessageHandlerFactory, ILoggerFactory loggerFactory,
+            IServiceScopeFactory serviceScopeFactory)
+                : base(bootstrapServers, eventuateKafkaConsumerConfigurationProperties, decoratedMessageHandlerFactory,
+                loggerFactory, serviceScopeFactory)
+            {
+
+            }
+        }
+        [TestMethod]
+        public void SendMessage()
+        {
+            IdGenerator generator = new IdGenerator();
+            var options = new DbContextOptionsBuilder<EventuateTramDbContext>().UseSqlServer(TestSettings.EventuateTramDbConnection).Options;
+            var schema = new EventuateSchema(TestSettings.EventuateTramDbSchema);
+            EventuateTramDbContextProvider provider = new EventuateTramDbContextProvider(options, schema);
+            MessageProducer mp = new MessageProducer(new List<IMessageInterceptor>(),
+            generator, provider, Substitute.For<ILogger<DatabaseMessageProducer>>());
+            mp.Send(destination, MessageBuilder.WithPayload(payload).Build());
+
+            Assert.IsNotNull(destination);
+        }
+
+        [TestMethod]
+        public void SubscribeMessage()
+        {
+            PrePostHandlerMessageHandlerDecorator obj = new PrePostHandlerMessageHandlerDecorator();
```

Comment:
You can pass in an empty list of decorators.

Reply:
Getting below exception if I pass empty list of decorators.
_Test method IO.Eventuate.Tram.Tests.MessageTest.ShouldReceiveMessage threw exception: 
    System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')_